### PR TITLE
This makes the comments page fetch hard coded comments from the server and add them to the DOM on page load

### PR DESCRIPTION
**Pull request by Aradhya-B**
- Added a dependency to Gson which makes it easy to convert objects to JSON in Java.
- Hard coded an array list of comments that are converted to JSON and served to comments page when the comments page loads
- The comments page creates list elements for each comment served as JSON and adds them to the DOM
image

### CODE REVIEW

File: portfolio/src/main/webapp/scripts/comments.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Fetches data from java servlet and adds it to DOM tree
+ */
+async function fetchAndInsertDataIntoDOM() {
+	const response = await fetch('/data');
+	const comments = await response.json();
+	const commentsListElement = document.getElementById('comments-list')
```
Comment (nickKelman) [inline portfolio/src/main/webapp/scripts/comments.js]: missing semi-colon

File: portfolio/src/main/webapp/scripts/comments.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	commentsListElement.innerHTML = '';
```
Comment (nickKelman) [inline portfolio/src/main/webapp/scripts/comments.js:8]: Why is this necessary?

**Review by nickKelman — APPROVED** [review]

File: portfolio/src/main/webapp/scripts/comments.js (hunk 1 above)
Comment (Aradhya-B) [reply portfolio/src/main/webapp/scripts/comments.js]: Got it.

File: portfolio/src/main/webapp/scripts/comments.js (hunk 2 above)
Comment (Aradhya-B) [reply portfolio/src/main/webapp/scripts/comments.js:8]: It's not really, just a safety check in case there was any hardcoded html in the commentsListElement.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,19 +14,33 @@
...
-    response.setContentType("text/html");
-    response.getWriter().println("Hello Aradhya!");
+    String json = convertArrayListToJson(comments);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  private String convertArrayListToJson(ArrayList<String> comments) {
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
```
Comment (cftorres) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: you could just return this instead of saving it in a variable because you don't ever need to reuse it

**Review by cftorres — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (Aradhya-B) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Good point, I copied this from the tutorial where I assume they were being as verbose as possible. Will make that change.
